app/Main.hs

```diff
@@ -51,35 +55,45 @@ unsafePathToText p = case T.toText p of
   Left t -> t
   Right t -> t
 
-checkFiles :: [T.FilePath] -> IO ()
-checkFiles = T.void . traverse checkFile
-  where
-    checkFile p = do
-      hasFile <- T.testfile p
-      T.when hasFile $
-        T.die $ "Found " <> unsafePathToText p <> ": there's already a project here. "
-             <> "Run `spacchetti local-setup --force` if you're sure you want to overwrite it."
-
 localSetup :: Bool -> IO ()
 localSetup force = do
-  T.unless force $
-    checkFiles [ pscPackageJsonPath, packagesDhallPath ]
-
-  T.touch pscPackageJsonPath
+  -- packages.dhall file
+  T.unless force $ do
+    hasDhall <- T.testfile packagesDhallPath
+    T.when hasDhall $
+      T.die $ "Found " <> unsafePathToText packagesDhallPath <> ": there's already a project here. "
+           <> "Run `spacchetti local-setup --force` if you're sure you want to overwrite it."
   T.touch packagesDhallPath
+  T.writeTextFile packagesDhallPath Templates.packagesDhall
 
-  pwd <- T.pwd
-  let projectName = case T.toText $ T.filename pwd of
-        Left _ -> "my-project"
-        Right n -> n
+  -- psc-package.json file
+  hasPscPackage <- T.testfile pscPackageJsonPath
+  if hasPscPackage && not force
+    then do
+      pscPackage <- T.readTextFile pscPackageJsonPath
+      case eitherDecodeStrict $ Text.encodeUtf8 pscPackage of
+        Left e -> T.die $ "The existing psc-package.json file is in the wrong format: " <>
+          Text.pack e
+        Right p -> do
+          T.writeTextFile pscPackageJsonPath $
+            Templates.encodePscPackage $ p { Templates.set = "local", Templates.source = "" }
+          T.echo "An existing psc-package.json file was found and upgraded to spacchetti."
```

Comment:
to use local package sets?

Reply:
What do you mean?

Reply:
"was found and upgraded to use local package sets"